pennylane/beta/plugins/default_tensor.py

```diff
@@ -568,77 +633,92 @@ def _get_operator_matrix(self, operation, par):
             return self._array(A, dtype=self.C_DTYPE)
         return self._asarray(A(*par), dtype=self.C_DTYPE)
 
-    def ev(self, obs_nodes, wires):
+    def ev(self, obs_nodes, wires, contraction_method="auto"):
         r"""Expectation value of observables on specified wires.
 
          Args:
-            obs_nodes (Sequence[tn.Node]): the observables as tensornetwork Nodes
+            obs_nodes (Sequence[tn.Node]): the observables as TensorNetwork Nodes
             wires (Sequence[Sequence[int]]): measured subsystems for each observable
+            contraction_method (str): The contraction method to be employed.
+                Possible choices are "auto", "greedy", "branch", or "optimal".
+                See TensorNetwork library documentation for more details.
          Returns:
             float: expectation value :math:`\expect{A} = \bra{\psi}A\ket{\psi}`
         """
-
-        all_wires = tuple(w for w in range(self.num_wires))
-        ket = self._add_node(self._state_node, wires=all_wires, name="Ket")
-        bra = self._add_node(tn.conj(ket), wires=all_wires, name="Bra")
-        meas_wires = []
-        # We need to build up <psi|A|psi> step-by-step.
-        # For wires which are measured, we need to connect edges between
-        # bra, obs_node, and ket.
-        # For wires which are not measured, we need to connect edges between
-        # bra and ket.
-        # We use the convention that the indices of a tensor are ordered like
-        # [output_idx1, output_idx2, ..., input_idx1, input_idx2, ...]
-        for obs_node, obs_wires in zip(obs_nodes, wires):
-            meas_wires.extend(obs_wires)
-            for idx, w in enumerate(obs_wires):
-                output_idx = idx
-                input_idx = len(obs_wires) + idx
-                self._add_edge(obs_node, input_idx, ket, w)  # A|psi>
-                self._add_edge(bra, w, obs_node, output_idx)  # <psi|A
-        for w in set(all_wires) - set(meas_wires):
-            self._add_edge(bra, w, ket, w)  # |psi[w]|**2
-
-        # At this stage, all nodes are connected, and the contraction yields a
-        # scalar value.
-        contracted_ket = ket
-        for obs_node in obs_nodes:
-            contracted_ket = tn.contract_between(obs_node, contracted_ket)
-        expval = tn.contract_between(bra, contracted_ket).tensor
-        if self._abs(self._imag(expval)) > tolerance:
+        if self._rep == "exact":
+            self._contract_to_ket(method=contraction_method)
+            ket = self._nodes["contracted_state"]
+            bra = tn.conj(ket, name="Bra")
```

Comment:
Just to make sure: naming would be similar to the previous names, right?

Reply:
You mean like `QubitStateVector`, `PauliX(0,)`, etc.? If so, then yes. But note that these names are mainly for debugging, so I'm trying not to worry about the formatting so much

Reply:
Oh I see! Sure, thanks!